src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -296,6 +293,16 @@ public synchronized IndexShard createShard(int sShardId, boolean primary) throws
         boolean success = false;
         Injector shardInjector = null;
         try {
+
+            ShardPath path = ShardPath.loadShardPath(logger, nodeEnv, shardId, indexSettings);
```

Comment:
this might end up throwing an  ElasticsearchIllegalStateException if there is an existing left over shard state but with the wrong uuid, causing the shard creation to fail if there is a local copy of an old shard (we used to safely ignore it).  It's a minor issue and we agreed to open a follow up issue.